lib/Polaris.Class.ps1

```diff
@@ -2,25 +2,140 @@ class Polaris {
     
     [Int]$Port
     [System.Collections.Generic.List[PolarisMiddleWare]]$RouteMiddleWare = [System.Collections.Generic.List[PolarisMiddleWare]]::new()
-    [System.Collections.Generic.Dictionary[string, [System.Collections.Generic.Dictionary[string, string]]]]$ScriptBlockRoutes = [System.Collections.Generic.Dictionary[string, [System.Collections.Generic.Dictionary[string, string]]]]::new()
+    [System.Collections.Generic.Dictionary[string, [System.Collections.Generic.Dictionary[string, scriptblock]]]]$ScriptBlockRoutes = [System.Collections.Generic.Dictionary[string, [System.Collections.Generic.Dictionary[string, scriptblock]]]]::new()
     hidden [Action[string]]$Logger
     hidden [System.Net.HttpListener]$Listener
-    hidden [System.Management.Automation.Runspaces.RunspacePool]$PowerShellPool = [RunspaceFactory]::CreateRunspacePool()
     hidden [bool]$StopServer = $False
     [string]$getLogsString = "PolarisLogs"
     [string] $ClassDefinitions = $script:ClassDefinitions
+    $ContextHandler = (New-ScriptBlockCallback -Callback {
+
+            param(
+                [System.IAsyncResult]
+                $AsyncResult
+            ) 
+
+            [Polaris]$Polaris = $AsyncResult.AsyncState
+            $context = $Polaris.Listener.EndGetContext($AsyncResult)
+        
+
+            if ($Polaris.StopServer -or $null -eq $context) {
+                if ($null -ne $Polaris.Listener) {
+                    $Polaris.Listener.Close()
+                }
+                break
+            }
+
+            $Polaris.Listener.BeginGetContext($Polaris.ContextHandler, $Polaris)
+
+            [System.Net.HttpListenerRequest] $rawRequest = $context.Request
+            [System.Net.HttpListenerResponse] $rawResponse = $context.Response
+
+            $Polaris.Log("request came in: " + $rawRequest.HttpMethod + " " + $rawRequest.RawUrl)
+
+            [PolarisRequest] $request = [PolarisRequest]::new($rawRequest)
+            [PolarisResponse] $response = [PolarisResponse]::new()
+
+            
+            [string] $route = $rawRequest.Url.AbsolutePath.TrimEnd('/')
+
+            if ( [string]::IsNullOrEmpty($route) ) { $route = "/" }
+
+            try {
+
+                # Run middleware in the order in which it was added
+                foreach ($middleware in $Polaris.RouteMiddleware) {
+                    $ScriptBlock = [scriptblock]::Create(
+                        "param(`$request,`$Response)`r`n" +
```

Comment:
not that it really matters but can you fix the casing on $request and $Response to be consistent?